Added documentation for qr code component (#1748)
OttoWinter/esphomedocs: components/display/index.rst

    - Axis labels are currently not possible without manually placing them.
    - The grid and border color is set with it.graph(), while the traces are defined separately.

QR Codes
********

Use this component to generate a QR-code containing a string on the device, which can then be drawn on compatible displays.

.. code-block:: yaml

    qr_code:
      - id: homepage_qr
        value: esphome.io

Configuration variables:

- **id** (**Required**, :ref:`config-id`): The ID with which you will be able to reference the QR-code later
  in your display code.
- **value** (**Required**, string): The string which you want to encode in the QR-code.
- **ecc** (*Optional*, string): The error correction code level you want to use. Defaults to ``LOW``. You can use one of the following values:

  - ``LOW`` - The QR Code can tolerate about 7% erroneous codewords
  - ``MEDIUM`` - The QR Code can tolerate about 15% erroneous codewords
  - ``QUARTILE`` - The QR Code can tolerate about 25% erroneous codewords
  - ``HIGH`` - The QR Code can tolerate about 30% erroneous codewords

> To draw the QR-code, call the `it.qr_code` function from your render lambda:

Reviewer: My mistake, this should've been double quoted:
```suggestion
To draw the QR-code, call the ``it.qr_code`` function from your render lambda:
```

    - Axis labels are currently not possible without manually placing them.
    - The grid and border color is set with it.graph(), while the traces are defined separately.

QR Codes
********

> Use this component to generate a qr code containing a string on the device, you can then draw the qr code on compatible displays.

Reviewer: ```suggestion
Use this component to generate a QR-code containing a string on the device, which can then be drawn on compatible displays.
```

It might be nice to explain which displays are compatible as well.

.. code-block:: yaml

    qr_code:
      - id: homepage_qr
        value: esphome.io

Configuration variables:

> - **id** (**Required**, :ref:`config-id`): The ID with which you will be able to reference the graph later

Reviewer: ```suggestion
- **id** (**Required**, :ref:`config-id`): The ID with which you will be able to reference the QR-code later
```

  in your display code.

> - **value** (**Required**, string): The string which you want to encode in the qr code.

Reviewer: ```suggestion
- **value** (**Required**, string): The string which you want to encode in the QR-code.
```

> - **scale** (*Optional*, int): The scale you want to use. Defaults to ``1``.

Reviewer: This option doesn't seem to exist:
```suggestion
```

> - **ecc** (*Optional*, string): The error correction code level you want oto use. Defaults to ``LOW``. You can use one of the following

Reviewer: ```suggestion
- **ecc** (*Optional*, string): The error correction code level you want to use. Defaults to ``LOW``. You can use one of the following values:
```

  - ``LOW`` - The QR Code can tolerate about 7% erroneous codewords
  - ``MEDIUM`` - The QR Code can tolerate about 15% erroneous codewords
  - ``QUARTILE`` - The QR Code can tolerate about 25% erroneous codewords
  - ``HIGH`` - The QR Code can tolerate about 30% erroneous codewords

> And then later in code:

Reviewer: ```suggestion
To draw the QR-code, call the `it.qr_code` function from your render lambda:
```
